action/protocol/staking/read_state.go

```diff
@@ -23,6 +26,17 @@ func toIoTeXTypesVoteBucketList(buckets []*VoteBucket) (*iotextypes.VoteBucketLi
 		if err != nil {
 			return nil, err
 		}
+		// fill in the endorsement
+		if b.isNative() {
+			endorsement, err := esr.Get(b.Index)
+			switch errors.Cause(err) {
+			case nil:
+				typBucket.EndorsementExpireBlockHeight = endorsement.ExpireHeight
+			case state.ErrStateNotExist:
+			default:
+				return nil, err
```

Comment:
after the withdrawal is success (i.e., chain get pass the expire height), you can still read the `esr` to get this bucket (and expire height), is that right?

Reply:
yes, withdrawal only update the `expireHeight`, and will not delete endorsement record